src/Ocelot/Configuration/Validator/FileConfigurationFluentValidator.cs

```diff
@@ -101,6 +109,24 @@ private static bool IsPlaceholderNotDuplicatedIn(string upstreamPathTemplate)
             var upstreamPathPlaceholders = matches.Select(m => m.Value);
             return upstreamPathPlaceholders.Count() == upstreamPathPlaceholders.Distinct().Count();
         }
+
+        private static bool SameNamePlaceholdersInUpstreamandDownstream(FileRoute route)
```

Comment:
@raman-m this breaks multiple tests is this ok ?

Reply:
But this is your code! Why are you asking me? ))

Reply:
@raman-m Sorry If I was not clear. It seems there are other unit tests that assume that the same Placeholder shouldn't exist in both upstream and downstream, should I change those unit tests because we are changing this feature ?

Reply:
Yes, we should!
But what tests are you talking about?
Just convert old tests to a `Theory` and add new `InlineData` for the bug scenario. And, that's it!

Reply:
@raman-m you can check the failing tests from Circle CI these are the tests that I left failing

Reply:
Are you stuck in fixing?
👌 I will check the failed build tomorrow...

Reply:
@AlyHKafoury  🆗 I've checked the 4 failed tests... They are related to service discovery for ServiceFabric etc...
So, DO you want to say me that our new validator breaks all these failed scenarios because the rule require the second placeholder od the pair?

Reply:
Correct, are we going to change the documentation and break compatibility and change the unit tests or we are gonna remove the code I added ?

Reply:
I don't know... 
You & I have to checkout feature branch and review the tests for carefully...
And we need to make final decision later... 
I guess, we can make exception for ServiceFabric routes, but I'm not sure they are real ServiceFabric ones...

Reply:
What I think about is that we allow the following : 
downstream: /bar/foo/
upstream: /foo/{placholder}/bar

but disallow the following : 
downstream: /foo/{placholder}/bar
upstream: /bar/foo/

Reply:
>@AlyHKafoury [on Jan 31](https://github.com/ThreeMammals/Ocelot/pull/1927#discussion_r1472732952)

Did you mean?
>upstream: /foo/{placeholder}/bar
>downstream: /bar/foo/{placeholder}

As far as I know, we don't allow empty placeholders in the middle of the path...
I haven't seen such logic... But we allow empty Catch All

----
>upstream: /foo/{placeholder}/bar

It doesn't make sense because when empty placeholder then this  is border case of `/foo/bar` URL path and better to define static route with this upstream path.

Allowing empty placeholders in the middle requires huge refactoring... I don't see much sense in this feature.

Reply:
@AlyHKafoury Today I understood finally your concern regarding Service Fabric tests failing! 🤩 
So, I've updated the validators and fixed tests in commit https://github.com/ThreeMammals/Ocelot/pull/1927/commits/643af3c09e1f9877a1cb4f483e8b18903b4d9b0b ✔️